downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -53,11 +53,12 @@ private fun findGitOrSubmoduleDir(workingDirOrFile: File): Repository {
 open class GitWorkingTree(workingDir: File, vcsType: VcsType) : WorkingTree(workingDir, vcsType) {
     companion object : Logging
 
-    private val repo = findGitOrSubmoduleDir(workingDir)
+    private val repo: Repository
+        get() = findGitOrSubmoduleDir(workingDir)
 
-    override fun isValid(): Boolean = repo.objectDatabase?.exists() == true
+    override fun isValid(): Boolean = repo.use { it.objectDatabase?.exists() == true }
```

Comment:
Commit-msg: The Repository class / interface seems to have an internal `useCount`, with the initial value of 1.
The class seems to release the resources when the count reaches 0. 

From the commit message I do not understand why the releasing of the resources is now triggered.
In particular what the actual trigger now is. Can you re-phrase it to explain that?
In particular, it should cover why the trigger is guaranteed to take place before the mentioned test clean-ups.